patches/server/1047-add-disguise-api.patch

```diff
@@ -0,0 +1,274 @@
++import net.minecraft.world.entity.player.Player;
++import net.minecraft.world.phys.Vec3;
++import org.bukkit.Bukkit;
++
++public final class DisguiseUtil {
++    private DisguiseUtil(){}
++
++    public static boolean tryDisguise(ServerPlayer player, Entity entity, ClientboundAddEntityPacket clientboundAddEntityPacket) {
++            return switch (entity.getBukkitEntity().getDisguiseData()) {
++                case DisguiseData.OriginalDisguise disguise -> false;
++                case io.papermc.paper.disguise.EntityTypeDisguise(var type) -> {
++                    player.connection.send(create(clientboundAddEntityPacket, org.bukkit.craftbukkit.entity.CraftEntityType.bukkitToMinecraft(type)));
++                    yield  true;
++                }
++                case io.papermc.paper.disguise.PlayerDisguise(var playerProfile, var listed, var skinParts) -> {
++                    PlayerProfile adapted = Bukkit.createProfile(entity.getUUID(), entity.hasCustomName()? entity.getBukkitEntity().getCustomName(): playerProfile.getName());
++                    adapted.setProperties(playerProfile.getProperties());
++                    net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Entry playerUpdate =
++                        new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Entry(
++                            entity.getUUID(),
++                            com.destroystokyo.paper.profile.CraftPlayerProfile.asAuthlibCopy(adapted),
++                            listed,
++                            0,
++                            net.minecraft.world.level.GameType.DEFAULT_MODE,
++                            entity.getCustomName(),
++                            null
++                        );
++                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket(EnumSet.of(net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket.Action.ADD_PLAYER), playerUpdate));
++                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LISTED), playerUpdate));
++                    player.connection.send(create(clientboundAddEntityPacket, net.minecraft.world.entity.EntityType.PLAYER));
++                    if(skinParts != null) {
++                        player.connection.send(new net.minecraft.network.protocol.game.ClientboundSetEntityDataPacket(
++                            clientboundAddEntityPacket.getId(),
++                            List.of(new SynchedEntityData.DataItem<>(Player.DATA_PLAYER_MODE_CUSTOMISATION, (byte) skinParts.getRaw()).value())
++                        ));
++                    }
++                    yield  true;
++                }
++            };
++    }
++
++    private static ClientboundAddEntityPacket create(ClientboundAddEntityPacket clientboundAddEntityPacket, EntityType<?> entityType) {
```

Comment:
MAJOR nit pick but if you name the arugment packet the method looks a whole lot nicer, lol